pilot/pkg/features/pilot.go

```diff
@@ -191,6 +191,12 @@ var (
 		"If enabled, protocol sniffing will be used for inbound listeners whose port protocol is not specified or unsupported",
 	)
 
+	EnableTCPMetadataExchange = env.RegisterBoolVar(
+		"PILOT_ENABLE_TCP_METADATA_EXCHANGE",
+		true,
```

Comment:
So it's true by default in 1.5? Could it cause troubles for upgrade path? (some servers have switched to the new ALPN but others might haven't)

Reply:
It should not be because, then ALPN won't match and metadata exchange won't get enabled.

Reply:
I mean if the server side was enabled, but some clients haven't, then does traffic will get into the (catch-all) filter chain? It my understanding, just want to confirm.

Reply:
So this is for TCP only.. so it should get into TCP specific filter chain only...
@yxue : that sound right, right?